File: 03/3.c
#include <stdio.h>
#include <math.h>

int isPrime(long long n);

int main()
{
	long long i , num = 600851475143 , largest;
	for(i = 3 ; i <= pow(num , 0.5) ; i+=2)
	{
		if (num % i == 0 && isPrime(i))
			largest = i;
	}
	printf("%lld\n" , largest);
}

int isPrime(long long n)
{
	long long i;
	if (n < 4)
		return 1;
	if (n%2 == 0)
		return 0;
	for(i = 3 ; i <= pow(n , 0.5) ; i += 2)
	{
		if(n % i == 0)
			return 0;
	}
	return 1;
}

File: 03/3.py
import math

def isPrime(n):
	if (n < 4):
		return True;
	for i in range(3 , int(n**(0.5)) + 1 , 2):
		if (n%i == 0):
			return False
	return True

def main():
	num = 600851475143
	for i in range(3 , int(num**(0.5)) + 1 , 2):
		if (num % i == 0 and isPrime(i)):
			print(i)

main()        
